src/cosim/serialization.cpp

```diff
@@ -1,11 +1,22 @@
 #include "cosim/serialization.hpp"
 
-#include <algorithm>
 #include <ios>
 #include <iterator>
 #include <utility>
+#include <cosim/log/logger.hpp>
+#include <iostream>
+#include <optional>
+#include <unordered_map>
+#include <cbor.h>
```

Comment:
These includes are not in the correct order, nor are they grouped appropriately. The correct order/format is:
```cpp
// First, main header for this `.cpp` file
#include "cosim/serialization.hpp"

// Then other libcosim headers (using double quotes in library-internal `.cpp` files)
#include "cosim/log/logger.hpp"

// Then third-party libraries
#include <cbor.h>

// Then stdlib headers
#include <ios>
#include <iostream>
#include <iterator>
#include <optional>
#include <unordered_map>
#include <utility>
```
Alphabetical order should be used within groups.

Reply:
Fixed. But it seems `logger.cpp/hpp` is a public header/source.

Reply:
Yeah, and this is something you'll see people having different opinions about. But the rule we've used thus far is to always use `#include <cosim/...>` in public headers (i.e., anything under `include/`), since those should *only* depend on other public headers, and to always use `#include "cosim/..."` in `.cpp` files, since they *may* pull in private headers.

One *could* imagine using angle brackets when including public headers and quotes when including private headers in `.cpp` files, but that would be too much of a hassle. Most of us probably don't remember which is which and would have to check every time. It's easier this way.